include/swift/AST/IRGenOptions.h

```diff
@@ -168,6 +173,7 @@ class IRGenOptions {
     Hash = (Hash << 1) | Optimize;
     Hash = (Hash << 1) | DisableLLVMOptzns;
     Hash = (Hash << 1) | DisableLLVMARCOpts;
+    Hash = (Hash << 2) | (unsigned)Sanitize;
```

Comment:
My understanding is that this computes the IR hash, which should be different for ASanified and non-ASanified code. ASan passes are just like optimization passes.